Mediapipe.Net.Examples.OsuFrameworkVisualTests/MediapipeDrawable.cs

```diff
@@ -46,53 +41,38 @@ public MediapipeDrawable(int cameraIndex = 0)
             });
         }
 
-        public void Start()
+#pragma warning disable IDE0051
+        [BackgroundDependencyLoader]
+        private void load(Camera camera, FrameConverter converter, FaceMeshCpuCalculator calculator)
         {
-            calculator.Run();
-            Camera.StartCapture();
+            this.camera = camera;
+            this.converter = converter;
+            this.calculator = calculator;
         }
+#pragma warning restore IDE0051
 
-        private unsafe void onFrame(object? sender, FrameEventArgs e)
+        protected override unsafe void Update()
         {
-            var frame = e.Frame;
-            if (converter == null)
-                converter = new FrameConverter(frame, PixelFormat.Rgba);
+            base.Update();
+            if (camera == null || converter == null || calculator == null)
+                return;
 
-            // Don't use a frame if it's not new
-            if (e.Status != DecodeStatus.NewFrame)
+            if (camera.TryGetFrame(out Frame frame) != DecodeStatus.NewFrame)
                 return;
 
             Frame cFrame = converter.Convert(frame);
-
-            ImageFrame imgFrame;
-            fixed (byte* rawDataPtr = cFrame.RawData)
-            {
-                imgFrame = new ImageFrame(ImageFormat.Srgba, cFrame.Width, cFrame.Height, cFrame.WidthStep,
-                    rawDataPtr);
-            }
-
+            ImageFrame imgFrame = new ImageFrame(ImageFormat.Srgba,
+                cFrame.Width, cFrame.Height, cFrame.WidthStep, cFrame.RawData);
             using ImageFrame outImgFrame = calculator.Send(imgFrame);
             imgFrame.Dispose();
 
             var span = new ReadOnlySpan<byte>(outImgFrame.MutablePixelData, outImgFrame.Height * outImgFrame.WidthStep);
-            var pixelData = SixLabors.ImageSharp.Image.LoadPixelData<Rgba32>(span, cFrame.Width, cFrame.Height);
+            var pixelData = Image.LoadPixelData<Rgba32>(span, cFrame.Width, cFrame.Height);
 
             texture ??= new Texture(cFrame.Width, cFrame.Height);
             texture.SetData(new TextureUpload(pixelData));
             sprite.FillAspectRatio = (float)cFrame.Width / cFrame.Height;
             sprite.Texture = texture;
         }
-
-        public new void Dispose()
```

Comment:
Why is the Dispose function removed in this Drawable?

Reply:
I had to circumvent around it and instead move the disposable dependencies of the drawable to the game base, because this method wasn't even being called.